lib/matplotlib/colorizer.py

```diff
@@ -232,9 +220,14 @@ def _set_cmap(self, cmap):
         cmap : `.Colormap` or str or None
         """
         # bury import to avoid circular imports
-        from matplotlib import cm
         in_init = self._cmap is None
-        self._cmap = cm._ensure_cmap(cmap)
+        cmap_obj = _ensure_cmap(cmap, accept_multivariate=True)
+        if not in_init:
+            if self.norm.n_components != cmap_obj.n_variates:
```

Comment:
Can join unless you expect to add more here.
```suggestion
        if not in_init and self.norm.n_components != cmap_obj.n_variates:
```